src/bin/report_collector.rs

```diff
@@ -282,5 +279,23 @@ async fn ipa(
 
     tracing::info!("{m:?}", m = ipa_query_config);
 
-    validate(expected, actual)
+    validate(&expected, &actual);
+
+    if let Some(ref path) = args.output_file {
```

Comment:
We should also record the QueryConfig and time taken for the run in the output file so that we have everything in a place

Reply:
I can do that probably, this requires a scheme for the output file, but it should not be too complicated to do